src/jdk.compiler/share/classes/com/sun/tools/javac/tree/TreeInfo.java

```diff
@@ -706,25 +707,26 @@ public static int positionFor(final Symbol sym, final JCTree tree) {
     /** Find the position for reporting an error about a symbol, where
      *  that symbol is defined somewhere in the given tree. */
     public static DiagnosticPosition diagnosticPositionFor(final Symbol sym, final JCTree tree) {
-        return diagnosticPositionFor(sym, tree, false);
+        return diagnosticPositionFor(sym, tree, null, false);
     }
 
-    public static DiagnosticPosition diagnosticPositionFor(final Symbol sym, final JCTree tree, boolean returnNullIfNotFound) {
+    public static DiagnosticPosition diagnosticPositionFor(final Symbol sym, final JCTree tree,
```

Comment:
I would keep the original declaration and add an overloaded one with the new argument so that only clients using the new declaration get affected